src/cr/cube/cubepart.py

```diff
@@ -616,11 +632,21 @@ def pairwise_significance_tests(self):
         significance test contains `p_vals` and `t_stats` (ndarrays that represent
         probability values and statistical scores).
         """
+        if self.has_overlaps:
+            return self._overlaps_significance_tests
+
         return tuple(
             PairwiseSignificance(self).values[column_idx]
             for column_idx in range(len(self.column_labels))
         )
 
+    @lazyproperty
+    def _overlaps_significance_tests(self):
+        return tuple(
+            SliceOverlapsSignificance(self).values[column_idx]
+            for column_idx in range(len(self.column_labels))
+        )
```

Comment:
I'd prefer to wrap this one into the `pairwise_significance_tests` and maybe have something like:
```python
if self.overlaps is not None:
    PairwiseSigClass = SliceOverlapsSignificance
else:
    PairwiseSigClass = PairwiseSignificance
return tuple(
            PairwiseSigClass(self).values[column_idx]
            for column_idx in range(len(self.column_labels))
        )
```